### PR TITLE
Extract filter parsing from `Query.parse` into `Filter.parse`

### PR DESCRIPTION
**Pull request by ggcampinho**
Refactor to allow new filters additions

### CODE REVIEW

**Comment by shaneog** [comment]
LGTM :+1:

**Comment by vnegrisolo** [comment]
LGTM too :+1:

File: lib/elasticband/query.rb [hunk 1]
```diff
@@ -136,7 +95,7 @@ def parse_boost_function(boost_options)
         if boost_options[:boost_by].present?
           ScoreFunction::FieldValueFactor.new(boost_options[:boost_by], modifier: :ln2p)
         elsif boost_options[:boost_where].present?
-          filter = join_filters(parse_filters(boost_options[:boost_where]))
+          filter = Filter.parse(only: boost_options[:boost_where])
           ScoreFunction::Filtered.new(filter, ScoreFunction::BoostFactor.new(1_000))
```
Comment (brennovich) [inline lib/elasticband/query.rb:99]: Not related with this PR, but is `1_000` a elasticsearch thing?

**Comment by brennovich** [comment]
:+1:

File: lib/elasticband/query.rb (hunk 1 above)
Comment (ggcampinho) [inline lib/elasticband/query.rb:99]: No, it's just a big boost based on searchkick. I think we can do it configurable later

**Pull request merged**